x/staking/keeper/liquid_stake_test.go

```diff
@@ -0,0 +1,514 @@
+
+	initialValidator := types.Validator{
+		OperatorAddress:          valAddress.String(),
+		TotalLiquidShares:        initialLiquidShares,
+		TotalValidatorBondShares: validatorBondShares,
+	}
+	app.StakingKeeper.SetValidator(ctx, initialValidator)
+
+	// Set validator bond factor so that the check is disabled
+	params := app.StakingKeeper.GetParams(ctx)
+	params.ValidatorBondFactor = sdk.NewDec(-1)
+	app.StakingKeeper.SetParams(ctx, params)
+
+	// Attempt to increase the validator liquid shares
+	// it should do nothing since the check is disabled
+	err := app.StakingKeeper.SafelyIncreaseValidatorTotalLiquidShares(ctx, initialValidator, increaseAmount)
+	require.NoError(t, err)
+
+	actualValidator, found := app.StakingKeeper.GetLiquidValidator(ctx, valAddress)
+	require.True(t, found)
+	require.Equal(t, initialLiquidShares, actualValidator.TotalLiquidShares, "shares with cap disabled")
+
+	// Change validator bond factor so that it is enabled, but the delegation will fail
+	params.ValidatorBondFactor = sdk.NewDec(1)
+	app.StakingKeeper.SetParams(ctx, params)
+
+	// Attempt to increase the validator liquid shares again, this time it should throw an
+	// error that the cap was exceeded
+	err = app.StakingKeeper.SafelyIncreaseValidatorTotalLiquidShares(ctx, initialValidator, increaseAmount)
+	require.ErrorIs(t, err, types.ErrInsufficientValidatorBondShares)
+	require.Equal(t, initialLiquidShares, actualValidator.TotalLiquidShares, "shares with strict cap")
+
+	// Change validator bond factor one more time, so that the increase succeeds
+	params.ValidatorBondFactor = sdk.NewDec(10)
+	app.StakingKeeper.SetParams(ctx, params)
+
+	// Finally, try the increase again and check that it succeeded
+	err = app.StakingKeeper.SafelyIncreaseValidatorTotalLiquidShares(ctx, initialValidator, increaseAmount)
+	require.NoError(t, err)
+
+	actualValidator, found = app.StakingKeeper.GetLiquidValidator(ctx, valAddress)
+	require.True(t, found)
+	require.Equal(t, initialLiquidShares.Add(increaseAmount), actualValidator.TotalLiquidShares, "shares with loose cap")
+}
+
+// Tests DecreaseValidatorTotalLiquidShares
+func TestDecreaseValidatorTotalLiquidShares(t *testing.T) {
+	_, app, ctx := createTestInput(t)
+
+	initialLiquidShares := sdk.NewDec(0)
+	decreaseAmount := sdk.NewDec(10)
+
+	// Create a validator with designated self-bond shares
+	privKey := secp256k1.GenPrivKey()
+	pubKey := privKey.PubKey()
+	valAddress := sdk.ValAddress(pubKey.Address())
+
+	initialValidator := types.Validator{
+		OperatorAddress:   valAddress.String(),
+		TotalLiquidShares: initialLiquidShares,
+	}
+	app.StakingKeeper.SetValidator(ctx, initialValidator)
+
+	// Set validator bond factor so that the check is disabled
+	params := app.StakingKeeper.GetParams(ctx)
+	params.ValidatorBondFactor = sdk.NewDec(-1)
+	app.StakingKeeper.SetParams(ctx, params)
+
+	// Attempt to decrease the validator liquid shares, it should not be changed since the cap is disabled
```

Comment:
What's the test doing here? 

Are you doing the following?
- Start with some liquid shares
- Disable cap
- Try to change liquid shares -> should fail because cap is turned off

Reply:
The `DecreaseLiquidShares` function has a check that ensures the cap is enabled before it starts doing accounting changes. This function is just checking that that works.

Slight correction to your summary:
* Starts with no liquid shares and disabled cap
* Attempt to change shares, the accounting changes should not occur
* Enable cap
* Attempt to change shares again, this time the total should change